src/v2/view-builder/utils/LinksUtil.js

```diff
@@ -30,32 +30,53 @@ const getSwitchAuthenticatorLink = (appState) => {
   return [];
 };
 
-const getForgotPasswordLink = (appState) => {
+const getForgotPasswordLink = (appState, settings) => {
```

Comment:
I think we can go advance here and just create one function that can give us all the links, we don't want to create a new function every time we add a new link.. What do you say ? We can pass different parameters to get different links ?

Reply:
That would be mega function. what's the benefit?

Reply:
I don't think it would be a mega function if you closely look at what we are doing in those functions. Isn't it ?

Reply:
I thought you meant merge 
```
getSwitchAuthenticatorLink,
getForgotPasswordLink,
goBackLink,
getSignOutLink
```
into one function that be able to generate all links base on parameter and it could be expanded over the course so I think it would be mega function.

Reply:
For more clarification, I intend to say that instead of exposing all these function we can expose just one. The LinksUtil can still keep these functions but those would be private. The single function that we expose can do the work of identifying required links for current view based on parameters.

Reply:
What would the extra parameter look like ?

Reply:
That single function still needs to be able to accept params that are needed to generate all the different types of links, even though you might just be trying to get `ForgotPasswordLink`. I am  not sure what that extra layer of abstraction adds here TBH when that top level function is internally going to call another method.  We might need different pieces of information to generate different types of links (appState, settings etc) and I feel its easier if have individual functions to handle those.

Reply:
this article shows the closest what i intend to achieve.

https://davidwalsh.name/javascript-functions

Reply:
Feels a little different. the example show how to use [higher order functions](https://en.wikipedia.org/wiki/Higher-order_function) but I thought what you meant is kind of Factory method. Similar to the ViewFactory which creates View per parameters. 
```
enum LinkType = Password | SignOut | ...

function createLink (appState, settings, LinkType) {
  case LinkType of
    Password        -> getPasswordLink(...),
    Signout            -> getSignOutLink(...),
    ...
}
```

It's not necessary a bad idea but just not sure if it's worth to introduce this abstract atm. :D